# Algorithms for Genetics

## Biology

### Gen 1: Sequencing by chain termination

Popular until ~2007

Human genome sequencing:
* used whole genome shotgun sequencing
* genome fragmented in 200-400kb
* fragments cloned in bacmid libraries
* libraries were fragmented
* Sanger sequencing performed
* then computational assembly

### Gen 2: High-throughput short read sequencing
* pyrosequencing
* sequencing by ligation
* nanoball sequencing
* ion torrent

### Gen 3: Single molecule long-read sequencing
* sequencing by synthesis (PacBio)
* Direct sequencing (Oxford Nanopore)
* Bleeding edge technology
  - high error rates (10-25%)

### Sequencing as a phenotyping tool
* CITE-seq (Stoeckius et al Nat Methods 2017)
* Spatial sequencing (Stahl et al Science 2016)

## Algorithms for Alignment - Burrows-Wheeler Transform

Suffix trie - data structure with fast lookup for determining substrings. Required space ~ space : O(n<sup>2</sup>)

Suffix tree - compressed suffix trie. Required space: O(20n)

Suffix array - only stores integers corresponding to each suffix. Required space space:  O(4n)

Seed Hash Tables -- indexing the human genome


Why the Burrow-Wheeler transform?
* suffix trees and suffix arrays require too much space
* useful for transformation

Burrows-Wheeler - 'reversible permutation' then sort
* puts runs of characters together
* encodes suffix array 
